src/components/forms/forms.js

```diff
@@ -13,80 +21,122 @@ import {
 
 import './forms.css';
 
+export const commonRequiredProps = ['id'];
+export const commonOptionalProps = [
+  'name',
+  'disabled',
+  'description',
+  'title',
+  'ariaLevel',
+  'defaultValue',
+  'hideTitle',
+  'style',
+  'validators',
+  'onChange',
+  'type'
+];
+
+// ----------------------------------------------- //
+// ======       MAIN FORM FIELD TYPES       ====== //
+// ----------------------------------------------- //
 export const FormFieldTypes = {
-  MULTITEXT: { defaultValue: [], component: formInputMultiText },
-  SLIDER: { defaultValue: false, component: formInputSlider },
-  RADIO: { defaultValue: null, component: formInputRadioGroup },
-  TEXT: { defaultValue: '', component: formInputGeneric },
-  NUMBER: { defaultValue: '', component: formInputGeneric },
+  MULTITEXT: {
+    defaultValue: [],
+    component: formInputMultiText,
+    requiredProps: [],
+    optionalProps: [
+      'placeholder',
+      'ariaDescribedby',
+      'inputStyle'
+    ],
+  },
+  SLIDER: {
+    defaultValue: false,
+    component: formInputSlider,
+    requiredProps: [],
+    optionalProps: [
+      'toggleText',
+    ],
+  },
+  RADIO: {
+    defaultValue: null,
+    component: formInputRadioGroup,
+    requiredProps: [
+      'options',
+      // 'options' example:
+      // [
+      //  {name: 'opt_1', text: 'Option 1'},
+      //  {name: 'other', text: 'Other', type: 'string'} <-- will have textbox
+      // ]
+    ],
+    optionalProps: [
+      'ariaDescribedBy',
+    ],
+    customPropValidation: customRadioPropValidation,
+  },
+  TEXT: {
+    defaultValue: '',
+    component: formInputGeneric,
+    requiredProps: [],
+    optionalProps: [
+      'placeholder',
+      'inputStyle',
+      'ariaDescribedby',
+    ],
+  },
+  NUMBER: {
+    defaultValue: '',
+    component: formInputGeneric,
+    requiredProps: [],
+    optionalProps: [
+      'placeholder',
+      'inputStyle',
+      'ariaDescribedby',
+    ],
+  },
   CHECKBOX: {
     defaultValue: (config) => (config?.valueType === 'string' ? '' : false),
-    component: formInputCheckbox
+    component: formInputCheckbox,
+    requiredProps: [],
+    optionalProps: [
+      'placeholder',
```

Comment:
Should checkboxes have a placeholder?

Reply:
these are used iff valueType is string; i will try to make this more clear!